File: eRxWeb/SPA/Model/select-medication/select-medication-search.model.ts
import { SelectMedicationRequestType } from '../../model/model.import.def';
export class SelectMedicationSearchModel {
    MedSearchText: string;
    MedicationSearchOption: SelectMedicationRequestType;
    PreBuiltGroupName: string;
    Validate: boolean = false;

    public constructor(medicationSearchText: string, medicationSearchOption: SelectMedicationRequestType, preBuiltGroupName: string) {
        this.MedSearchText = medicationSearchText;
        this.MedicationSearchOption = medicationSearchOption;
        this.PreBuiltGroupName = preBuiltGroupName;
    }
}

export class SelectMedicationSearchArgs {
    MedicationSearchCriteria: SelectMedicationSearchModel;
    public constructor(medSearchCriteria: SelectMedicationSearchModel) {
        this.MedicationSearchCriteria = medSearchCriteria;
    }
}
export enum PrescriptionTaskType {
    APPROVAL_REQUEST = 0,
    REJECTION_NOTIFICATION,
    RENEWAL_REQUEST,    //internal renewal
    SCRIPT_ERROR, //ERROR
    SEND_TO_ADMIN,
    REFREQ, //from pharmacy
    SEND_TO_DISPENSE,
    EPA, //electronic prior authorization
    RXCHG = 8,
    RXCHG_PRIORAUTH = 9, //Subtask of a ChangeRx
    SPECIALTY_MED_ASSIST_RX = 10, //SpecialtyMed, specialty med
    DEFAULT = 100 //Used as default value when casting
}
export class SelectMedicationSearchStartUpParameters {
    DoctorHistoryOptionDisplayText: string;
    IsShowPreBuiltGroup: boolean;
    IsSearchDisabled: boolean;
}
export class RetrieveTaskTypeParameters {
    PrescriptionTaskType: PrescriptionTaskType;
    IsCsRefReqWorkflow: boolean;
    IsCsRxChgWorkflow: boolean;
    IsReconcileREFREQNonCS: boolean;
}

File: eRxWeb/ServerModel/SelectMedicationSearchModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Allscripts.ePrescribe.Common;

namespace eRxWeb.ServerModel
{
    public class SelectMedicationSearchModel
    {
    }

    public class RetrieveTaskTypeParameters
    {
        public Constants.PrescriptionTaskType PrescriptionTaskType { get; set; }
        public bool IsCsRefReqWorkflow { get; set; }
        public bool IsCsRxChgWorkflow { get; set; }
        public bool IsReconcileREFREQNonCS { get; set; }
        public RetrieveTaskTypeParameters(Constants.PrescriptionTaskType taskType, bool isCsRefReqWorkflow, bool isCsRxChgWorkflow, bool isReconcileREFREQNonCS)
        {
            this.PrescriptionTaskType = taskType;
            this.IsCsRefReqWorkflow = isCsRefReqWorkflow;
            this.IsCsRxChgWorkflow = isCsRxChgWorkflow;
            this.IsReconcileREFREQNonCS = isReconcileREFREQNonCS;
        }
    }
    public class SelectMedicationSearchStartUpParameters
    {
        public string DoctorHistoryOptionDisplayText { get; set; }
        public bool IsShowPreBuiltGroup { get; set; }
        public bool IsSearchDisabled { get; set; }

        public SelectMedicationSearchStartUpParameters(string doctorHistoryOptionDisplayText, bool isShowPreBuiltGroup, bool isCSTask)
        {
            this.DoctorHistoryOptionDisplayText = doctorHistoryOptionDisplayText;
            this.IsShowPreBuiltGroup = isShowPreBuiltGroup;
            this.IsSearchDisabled = isCSTask;
        }
    }
}